modules/nw-enforcing-hsts-per-domain.adoc

```diff
@@ -0,0 +1,96 @@
+// Module included in the following assemblies:
+// * networking/configuring-routing.adoc
+
+[id="nw-enforcing-hsts-per-domain_{context}"]
+= Enforcing HTTP strict transport security per-domain
+
+To enforce HTTP strict transport security (HSTS) per-domain for all TLS routes, add a `requiredHSTSPolicies` record to the Ingress spec to capture the configuration of the HSTS policy.
+
+If you configure the `route.openshift.io/RequiredRouteAnnotations` plug-in to enforce HSTS, then any newly created route must be configured with a compliant HSTS policy, which is verified against the global setting on the cluster Ingress configuration, called `ingresses.config.openshift.io/cluster`.
+
+[NOTE]
+====
+To handle upgraded clusters with non-compliant HSTS routes, you can update the manifests at the source and apply the updates.
+====
+
+[IMPORTANT]
+====
+HSTS cannot be applied to non-TLS routes, even if HSTS is requested for all routes globally.
+====
+
+.Prerequisites
+
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
+* You installed the `oc` CLI.
+
+.Procedure
+
+. Edit the Ingress config file:
++
+[source,terminal]
+----
+$ oc edit ingresses.config.openshift.io/cluster
+----
++
+.Example HSTS policy
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Ingress
+metadata:
+  name: cluster
+spec:
+  domain: apps.abc.com
+  requiredHSTSPolicies: <1>
+  - domainPatterns: <2>
+    - abc.com
+    namespaceSelector: <3>
+      matchLabels:
+        myPolicy: strict
+    maxAge: <4>
+      smallestMaxAge: 1
+      largestMaxAge: 31536000
+    preloadPolicy: RequirePreload <5>
+    includeSubDomainsPolicy: RequireIncludeSubDomains
```

Comment:
We need a <6> for includeSubDomainsPolicy

Reply:
It can be left blank, or options are: RequireIncludeSubDomains, RequireNoIncludeSubDomains, or NoOpinion